src/Leela.cpp

```diff
@@ -222,6 +222,36 @@ static void parse_commandline(int argc, char *argv[], bool & gtp_mode) {
         }
     }
 #endif
+
+    auto out = std::stringstream{};
+    for (const auto& it : vm) {
+        if (it.second.defaulted()) {
+            continue;
+        }
+        auto &type = it.second.value().type();
+
+        out << "--" << it.first;
+        if (type == typeid(uint64)) {
+            out << " " << it.second.as<uint64>();
+        } else if (type == typeid(std::string)) {
+            // skip unary string options
+            if (it.second.as<std::string>() != "") {
+                // Sanitize user supplied strings
+                out << " " << it.first;
+            }
+        } else if (type == typeid(int)) {
+            out << " " << it.second.as<int>();
+        } else if (type == typeid(float)) {
+            out << " " << it.second.as<float>();
+        } else {
+            out << " unknown option type";
+        }
+        out << " ";
+    }
```

Comment:
What is the difference between this and collecting the original argv/argc?

Reply:
It was easier this way to know which options take string arguments, so I can protect user information. If I parse argv/argc the code has to be updated when new string arguments are added. Also the parser takes care of all the weird cases like extra spaces etc that can happen, what if the weights filename has a space in it, or some weird characters. I don't want to rewrite a parser for those cases. 

That said I can change it to parsing argv/argc if you want. Let me know.

Reply:
Maybe I'm paranoid but for example in Windows if you copy/paste a network in a directory you might have a command line like this:
leelaz.exe -w "..\154eaebd - Copy"

And parser has to handle quotes, backslashes, spaces, and something that looks like a -Copy command line argument correctly.

Reply:
Oh sorry one more thing, I can look into using the short options instead of the long ones (-n not --noise) if you prefer to be less verbose.

Reply:
After thinking it over, almost all use cases won't have user info like a directory, so maybe it's not worth doing all this. So I changed it to just use argv verbatim.